pwndbg/disasm/arch.py

```diff
@@ -503,6 +513,37 @@ def _telescope_format_list(self, addresses: List[int], limit: int, emu: Emulator
                 enhance_string_len=enhance_string_len,
             )
 
+    def _enhance_syscall(self, instruction: PwndbgInstruction, emu: Emulator) -> None:
+        if CS_GRP_INT not in instruction.groups:
+            return None
+
+        syscall_register = pwndbg.lib.abi.ABI.syscall().syscall_register
+        syscall_arch = pwndbg.gdblib.arch.name
+
+        # On x86/x64 `syscall` and `int <value>` instructions are in CS_GRP_INT
+        # but only `syscall` and `int 0x80` actually execute syscalls on Linux.
+        # So here, we return no syscall name for other instructions and we also
+        # handle a case when 32-bit syscalls are executed on x64
+        if syscall_register in ("eax", "rax"):
+            mnemonic = instruction.mnemonic
+
+            # We read .imm directly, because at this point we haven't enhanced the operands with values
+            is_32bit = mnemonic == "int" and instruction.operands[0].imm == 0x80
+            if not (mnemonic == "syscall" or is_32bit):
+                return None
+
+            # On x64 the int 0x80 instruction executes 32-bit syscalls from i386
+            # On x86, the syscall_arch is already i386, so its all fine
+            if is_32bit:
+                syscall_arch = "i386"
```

Comment:
Maybe we add a method in `DisassemblyAssistant` to get the syscall arch, the default implementation returns `pwndbg.gdblib.arch.name`, but the x86 assistant does this? That way you can make this method in the base class arch independent (without using if statements).

Reply:
That's clean - latest commit implements it this way.